plugin/types.go

```diff
@@ -58,6 +59,16 @@ type ClientHook interface {
 	Resource() client.Object
 }
 
+type Interceptor interface {
+	syncertypes.Base
+
+	// Handler is the handler that will handle the requests delegated by the syncer
+	Handler() http.Handler
```

Comment:
Can the intercepter not just implement http.Handler itself? Would make it easier to consume.